sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/MigrationGuideSnippetsLiveTests.cs

```diff
@@ -0,0 +1,192 @@
+                };
+
+                BlobClient blobClient = storageClient.GetBlobClient($"{ prefix }{ checkpoint.PartitionId }");
+
+                using var content = new MemoryStream(Array.Empty<byte>());
+                await blobClient.UploadAsync(content, metadata: metadata, cancellationToken: cancellationSource.Token);
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Reads and parses a set of fake legacy checkpoint
+        ///   data in JSON format.
+        /// </summary>
+        ///
+        /// <param name="=fakeContainerName">The name of a fake container; used only to avoid warnings in the calling code.</param>
+        ///
+        /// <returns>The set of fake legacy checkpoints.</returns>
+        ///
+        private IEnumerable<MigrationCheckpoint> ReadFakeLegacyCheckpoints(string fakeContainerName)
+        {
+            var checkpointJson = @"[
+            {
+                ""PartitionId"":""1"",
+                ""Owner"":""eecd42df-a253-49d1-bb04-e5f00c106cfc"",
+                ""Token"":""6271aadb-801f-4ec7-a011-a008808a656c"",
+                ""Epoch"":5,
+                ""Offset"":""400"",
+                ""SequenceNumber"":125
+            },
+            {
+                ""PartitionId"":""2"",
+                ""Owner"":""eecd42df-a253-49d1-bb04-e5f00c106cfc"",
+                ""Token"":""6271aadb-801f-4ec7-a011-a008808a656c"",
+                ""Epoch"":1,
+                ""Offset"":""78"",
+                ""SequenceNumber"":39
+            }]";
+
+            return JsonSerializer.Deserialize<MigrationCheckpoint[]>(checkpointJson);
+        }
+
+        #region Snippet:EventHubs_Migrate_LegacyCheckpoints
+
+        private async Task<List<MigrationCheckpoint>> ReadLegacyCheckpoints(
+            string connectionString,
+            string container,
+            string consumerGroup,
+            CancellationToken cancellationToken)
+        {
+            var storageClient = new BlobContainerClient(connectionString, container);
+
+            // If there is no container, no action can be taken.
+
+            if (!(await storageClient.ExistsAsync(cancellationToken)))
+            {
+                throw new ArgumentException("The source container does not exist.", nameof(container));
+            }
+
+            // Read and process the legacy checkpoints.
+
+            var checkpoints = new List<MigrationCheckpoint>();
+
+            await foreach (var blobItem in storageClient.GetBlobsAsync(BlobTraits.All, BlobStates.All, consumerGroup, cancellationToken))
+            {
+                using var blobContentStream = new MemoryStream();
+                await (storageClient.GetBlobClient(blobItem.Name)).DownloadToAsync(blobContentStream);
+
+                var checkpoint = JsonSerializer.Deserialize<MigrationCheckpoint>(Encoding.UTF8.GetString(blobContentStream.ToArray()));
```

Comment:
Could we use the `ReadOnlySpan<byte>` overloads rather than converting these bytes into a string? *has bad memories from OOM when trying to fit large blobs in memory*

Reply:
Not really, because the content is in a stream and need to have the entire set of content available to parse as JSON and deserialize.   You'd have to read the stream into a buffer, which would have the same cost as the array creation.   Unfortunately, there's no way to stream it in.   We could rent a shared buffer and return it after the deserialization, but that's not a widely-used optimization and I'd like to keep this approachable as a sample.

On the upside, we know that we're reading checkpoint data, which is no more than 2k, so we shouldn't see OOM conditions unless the machine is taxed elsewhere.